### ISSUE TITLE
NVDA does not track text selection when editing cells in Microsoft Excel

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2946]
**Reported by miahtech on 2013-01-26 05:30**
The summary pretty much says it all. If you are editing information in a cell, and try to select text, NVDA does not track the selection with speech or in Braille. 
**Blocked by #770**
**Blocking #5407**

**Comment by ehollig** [comment]
This appears to work correctly for me in Excel 2016. Can someone test using Excel 2010 to see if they get the same results? CC @bhavyashah

**Comment by bhavyashah** [comment]
Confirmed. Using NVDA 2017.2 and Excel 2010on a Windows 8.1 system, please use the following STR to successfully replicate the reported issue:
* Open a new Excel workbook. Your focus will most likely land on a1.
* Type 'hello' in a1.
* Although arrowing around 'hello' works fine, pressing Shift+left/right arrow results in silence.
I ensured that Shift+left/right arrow was indeed selecting the correct characters by pressing Ctrl+C and checking the clipboard via NVDA+C.
By no means would I consider this issue trivial. Thus, I suggesting marking as a P2 or P3. CC @michaelDCurran @feerrenrut

**Comment by bhavyashah** [comment]
So far, we know that this issue exists with Excel 2010 but not with Excel 2016. @Qchristensen Would you be able to test this with Excel 2013? @feerrenrut What other information do we require in order to determine a priority for this ticket?

**Comment by Qchristensen** [comment]
Using Excel 2016 / 365 - version 1807, build 10325.20082 and NVDA Alpha 15858.949f0bab NVDA is correctly reporting selected text when editing cells in Excel.

Note that as per issue #8146 I have "Allow editing in cells" unchecked (with that setting checked, which is the default, I can't read anything when editing cells, selected or not).

**Comment by Adriani90** [comment]
This is still reproducible with NVDA 2019.1.1 in Excel 2010. In Excel 2016 / 365 everything works correctly.
cc: @lukaszgo1, @leonardder

**Comment by Qchristensen** [comment]
In Excel 365, "Allow editing in cells" can now be checked or unchecked, it works either way.

I don't have Excel 2010, but in Excel 2007, this error still exists with NVDA 2019.1.1.  If you edit a cell, you can use the arrows to move through the text and it reads fine, but as soon as you press SHIFT+ARROW, the text gets selected, but NVDA does not report anything.  Press NVDA+shift+up arrow to report selected text and NVDA reports "no text selected".

**Comment by Adriani90** [comment]
End of life for MS Office 2013 is officially on April 11, 2023. Thus I am closing this issue as abandoned since all MS office product versions before 2016 will reach end of life by this date. If you have the same issue in later version of MS Office (beginning with MS office 2016), please open a new issue covering those versions.

**Comment by Adriani90** [comment]
Actually, as of MS Excel 365 2303 Build 16.0.16227.20202) it seems NVDA does not report what you select when editing a cell. This happens only with uIA disabled.

I am reopening this one.

**Comment by Qchristensen** [comment]
Confirming this is still an issue with
Excel in Office 365 (64-bit) Version: 16.0.16827.20166 running on
Windows 11 (64-bit) Version: 22H2, Build: 22621.2428 
using NVDA 2023.3

When the Excel option "Allow editing directly in cells" is checked.

1. Press alt+f, then t to open Excel's options
2. Press CONTROL+TAB to Advanced (7 times for me)
3. Press ALT+E to toggle "Allow editing directly in cells"  (With this option not checked, selected text is reported correctly).

**Comment by Stallon1999** [comment]
In the MS office 2021 also, this issue persist.

**Comment by Adriani90** [comment]
Closing as fixed by #16473.
